packages/frontend/src/components/dialogs/ProxyConfiguration/index.tsx

```diff
@@ -0,0 +1,403 @@
+      }
+    }
+    updateProxySettings()
+  }, [
+    proxyState.enabled,
+    accountId,
+    openAlertDialog,
+    tx,
+    proxyState.proxies,
+    proxyState.activeProxy,
+    proxyState.updateSettings,
+    proxyState,
+  ])
+
+  /**
+   * validate settings before closing
+   * to avoid invalid settings
+   */
+  const closeDialog = () => {
+    if (proxyState.enabled && proxyState.proxies.length === 0) {
+      openAlertDialog({
+        message: tx('proxy_invalid'),
+      })
+      return
+    }
+    onClose()
+  }
+
+  const copyToClipboard = useCallback(
+    (url: string) => {
+      navigator.clipboard.writeText(url)
+      openAlertDialog({
+        message: `${url}\n${tx('copied_to_clipboard')}`,
+      })
+    },
+    [tx, openAlertDialog]
+  )
+
+  // some basic validations, returns true if the url seems valid
+  // and is not already in the list of proxies
+  const maybeValidProxyUrl = (url: string): boolean => {
+    const parts = url.split('://')
+    return (
+      parts.length === 2 &&
+      parts[0].length >= 2 && // shortest protocol is ss://
+      parts[1].length >= 1 && // host
+      !proxyState.proxies.includes(url)
+    )
+  }
+
+  return (
+    <Dialog
+      fixed
+      width={400}
+      dataTestid='proxy-dialog'
+      canOutsideClickClose={false}
+    >
+      <DialogHeader
+        title={tx('menu_settings')}
+        onClose={closeDialog}
+        dataTestid='proxy-settings'
+      />
+      <DialogBody className={styles.proxyDialogBody}>
+        <div className={styles.container}>
+          <SettingsSwitch
+            label={tx('proxy_use_proxy')}
+            value={proxyState.enabled}
+            onChange={changeProxyEnable}
+          />
+          <div>
+            <h3 className='title'>{tx('proxy_list_header')}</h3>
+            <p className={styles.explain}>{tx('proxy_add_explain')}</p>
+          </div>
+          <div className={styles.proxyList} role='radiogroup'>
+            {proxyState.proxies.map((proxyUrl, index) => (
+              <ProxyItemRow
+                key={index}
+                proxyUrl={proxyUrl}
+                index={index}
+                isActive={proxyUrl === proxyState.activeProxy}
+                configured={configured}
+                changeActiveProxy={changeActiveProxy}
+                connectivityStatus={connectivityStatus}
+                copyToClipboard={copyToClipboard}
+                deleteProxy={deleteProxy}
+              />
+            ))}
+          </div>
+        </div>
+      </DialogBody>
+      <DialogFooter>
+        {showNewProxyForm && (
+          <form>
+            <DeltaInput
+              label={tx('proxy_add_url_hint')}
+              value={newProxyUrl}
+              onChange={e => setNewProxyUrl(e.target.value)}
+            />
+            <Button
+              className='save-proxy'
+              onClick={() => addProxy(newProxyUrl)}
+              styling='primary'
+              disabled={!maybeValidProxyUrl(newProxyUrl)}
```

Comment:
I think this should be a validation error on the proxy input (`setCustomValidity`) as opposed to the button being disabled with no indication of what's wrong.